akka-stream/src/main/scala/package.scala

```diff
@@ -22,6 +22,6 @@ package object akkastream {
     extends reactivemongo.api.CursorFlattener[AkkaStreamCursor] {
 
     def flatten[T](future: Future[AkkaStreamCursor[T]]): AkkaStreamCursor[T] =
-      new AkkaStreamFlattenedCursor(future)
+      cursorProducer[T].produce(Cursor.flatten[T, Cursor](future)(CursorFlattener.defaultCursorFlattener))
```

Comment:
I need to rollback this change, as `FlattenedCursor` does not (and cannot) implement `CursorOps`.
Ideally we would use `Source.fromFutureSource` in `AkkaStreamFlattenedCursor` to preserve the materialized value, but is not available in akka 2.4... Shoud we backport `fromFutureSource` or just continue to return an already completed future for flatten `documentSource`s ?

Reply:
Backporting `fromFutureSource` is not as trivial as it seems

Reply:
another possibility is to make `FlattenedCursor` implement `CursorOps` in a blocking fashion when required...?

Reply:
I prefer not to introduce blocking, knowing that since Akka 2.5 a clean/async solution is possible (even if for now it means not for Akka 2.4 compat).